src/odemis/gui/cont/fastem_acq.py

```diff
@@ -581,32 +563,231 @@ def _on_alignment_done(self, future):
+        (either successfully, cancelled or failed).
+        It stores the calibrated parameters on the ROC object.
+        :param future: (ProgressiveFuture) Calibration future object, which can be cancelled.
+        :param roc_num: (int) The number of the region of calibration.
+        """
+
+        try:
+            config = future.result()  # wait until the calibration is done
+            self.save_calibrated_settings(roc_num, config)  # save calibrated
+        except CancelledError:
+            self.is_aligned.value = False  # don't enable ROA acquisition
+            self._update_calibration_controls("Calibration cancelled")  # update label to indicate cancelling
+            # TODO @Eric is then the callback for all futures still called?
+            # return
+        except Exception as ex:
+            logging.exception("Calibration failed with %s.", ex)
+            # return
+
+    @call_in_wx_main  # as changes in gui are triggered
+    def _on_batch_calibrations_done(self, batch_future):
+        """
+        Called when all calibrations are finished (either successfully, cancelled or failed).
+        :param batch_future: (ProgressiveFuture) Calibration future object, which can be cancelled.
+        """
+
+        self._tab_data.main.is_acquiring.value = False
+        self._future_connector = None  # reset connection to the progress bar
+
+        try:
+            batch_future.result()  # wait until the calibration is done
+            self.is_aligned.value = True  # allow acquiring ROAs
+            self._update_calibration_controls("Calibration successful")
+        except CancelledError:
+            self.is_aligned.value = False  # don't enable ROA acquisition
+            self._update_calibration_controls("Calibration cancelled")  # update label to indicate cancelling
+            # return
+        except Exception as ex:
+            self.is_aligned.value = False  # don't enable ROA acquisition
+            logging.exception("Calibration failed with %s.", ex)
+            self._update_calibration_controls("Calibration failed")
+            # return
+
+    def _on_calibration_state(self, _=None):
+        """
+        Updates the calibration state based on the active (loaded) scintillators and whether
+        any regions of calibration (roc) are selected.
+        """
+        # check if any scintillators loaded
+        if not self._main_data_model.active_scintillators.value:
+            self._update_calibration_controls("No scintillator loaded (go to Chamber tab).", False)
         else:
-            duration = align.fastem.estimate_calibration_time(self.calibrations)
-            self.gauge_label.SetLabel(str(duration) + " seconds")
+            # check if at least one roc was selected
+            if all(roc.coordinates.value == stream.UNDEFINED_ROI
+                   for roc in self._tab_data.calibration_regions.value.values()):
```

Comment:
The comment does not match the actual code. By reading the comment I would think `all(roc.coordinates.value == stream.UNDEFINED_ROI for roc in self._tab_data.calibration_regions.value.values()) = True` would mean that at least one roc was selected, but if I read the code it means that no roc was selected. You could update it to:
```
            # check if at least one roc was selected
            if any(roc.coordinates.value != stream.UNDEFINED_ROI
                   for roc in self._tab_data.calibration_regions.value.values()):
                self._update_calibration_controls()
            else:
                self._update_calibration_controls("No calibration region selected.", False)
```